geth/jail/jail.go

```diff
@@ -46,17 +46,39 @@ func (jail *Jail) BaseJS(js string) {
 }
 
 // NewJailCell initializes and returns jail cell
-func (jail *Jail) NewJailCell(id string) common.JailCell {
+func (jail *Jail) NewJailCell(id string) (common.JailCell, error) {
+	if jail == nil {
+		return nil, ErrInvalidJail
+	}
+
 	vm := otto.New()
 
 	newJail, err := newJailCell(id, vm, loop.New(vm))
 	if err != nil {
-		//TODO(alex): Should we really panic here, his there
-		// a better way. Think on it.
-		panic(err)
+		return nil, err
+	}
+
+	jail.RLock()
```

Comment:
`jail.cells[id] = newJail` operation requires write lock. So you should do `jail.Lock()`. Also, putting it in a block is not necessary.